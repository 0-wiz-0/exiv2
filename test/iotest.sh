#!/usr/bin/env bash
# Test driver for image file i/o

source ./functions.source

(   cd "$testdir"

    errors=0
    test_files="table.jpg smiley2.jpg ext.dat"
    echo
    printf "file io tests"
    for i in $test_files; do ioTest $i; done

    printf "\n---------------------------------------------------------\n"
    if [ $errors -eq 0 ]; then
       echo 'All test cases passed'
    else
       echo $errors 'test case(s) failed!'
    fi
)

# Test http I/O
if [ "$EXIV2_PORT" != "None" -a "$EXIV2_HTTP" != "None" ]; then
    startHttpServer
    if [ ! -z $exiv2_httpServer ]; then
        (   cd "${testdir}" 
            >&2 printf "*** HTTP tests begin\n"

            cd "$testdir"
            test_files="table.jpg Reagan.tiff exiv2-bug922a.jpg"
            for i in $test_files; do
                runTest iotest s0 s1 s2                    $url/data/$i
                for t in s0 s1 s2 $url/data/$i; do
                    runTest exiv2 -g City -g DateTime $t
                done
            done
            # Format spec for stat
            F='-c%s'
            if [ $PLATFORM  == 'Darwin' -o $PLATFORM == 'NetBSD' -o $PLATFORM == 'FreeBSD' ]; then
            F='-f%z'
            fi
            
            runTest iotest s0 s1 s2 $url/data/table.jpg 1
            echo $(stat $F s0 s1 s2 ../data/table.jpg) $(checkSum s0) $(checkSum s1) $(checkSum s2) $(checkSum ../data/table.jpg)
            runTest iotest s0 s1 s2 $url/data/table.jpg 10
            echo $(stat $F s0 s1 s2 ../data/table.jpg) $(checkSum s0) $(checkSum s1) $(checkSum s2) $(checkSum ../data/table.jpg)
            runTest iotest s0 s1 s2 $url/data/table.jpg 1000
            echo $(stat $F s0 s1 s2 ../data/table.jpg) $(checkSum s0) $(checkSum s1) $(checkSum s2) $(checkSum ../data/table.jpg)
            >&2 printf "*** HTTP tests end\n"
        )  | tr -d '\r' | sed 's/[ \t]+$//' > $results
        reportTest
    fi
    closeHttpServer
fi

# That's all Folks!
##