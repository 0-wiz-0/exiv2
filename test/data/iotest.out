http://0.0.0.0:1276/data/table.jpg length = 568
s0                    Exif.Image.DateTime                          Ascii      21  2004-07-13T21:23:44Z
s0                    Iptc.Application2.City                       String      7  Seattle
s1                    Exif.Image.DateTime                          Ascii      21  2004-07-13T21:23:44Z
s1                    Iptc.Application2.City                       String      7  Seattle
s2                    Exif.Image.DateTime                          Ascii      21  2004-07-13T21:23:44Z
s2                    Iptc.Application2.City                       String      7  Seattle
http://0.0.0.0:1276/data/table.jpg  Exif.Image.DateTime                          Ascii      21  2004-07-13T21:23:44Z
http://0.0.0.0:1276/data/table.jpg  Iptc.Application2.City                       String      7  Seattle
http://0.0.0.0:1276/data/Reagan.tiff length = 114466
s0                    Exif.Image.DateTime                          Ascii      20  2012:04:07 16:11:27
s0                    Exif.Photo.DateTimeOriginal                  Ascii      20  2004:06:21 23:37:53
s0                    Exif.Photo.DateTimeDigitized                 Ascii      20  2004:06:21 23:37:53
s0                    Iptc.Application2.City                       String     19  Straits of Magellan
s0                    Xmp.photoshop.City                           XmpText    19  Straits of Magellan
s1                    Exif.Image.DateTime                          Ascii      20  2012:04:07 16:11:27
s1                    Exif.Photo.DateTimeOriginal                  Ascii      20  2004:06:21 23:37:53
s1                    Exif.Photo.DateTimeDigitized                 Ascii      20  2004:06:21 23:37:53
s1                    Iptc.Application2.City                       String     19  Straits of Magellan
s1                    Xmp.photoshop.City                           XmpText    19  Straits of Magellan
s2                    Exif.Image.DateTime                          Ascii      20  2012:04:07 16:11:27
s2                    Exif.Photo.DateTimeOriginal                  Ascii      20  2004:06:21 23:37:53
s2                    Exif.Photo.DateTimeDigitized                 Ascii      20  2004:06:21 23:37:53
s2                    Iptc.Application2.City                       String     19  Straits of Magellan
s2                    Xmp.photoshop.City                           XmpText    19  Straits of Magellan
http://0.0.0.0:1276/data/Reagan.tiff  Exif.Image.DateTime                          Ascii      20  2012:04:07 16:11:27
http://0.0.0.0:1276/data/Reagan.tiff  Exif.Photo.DateTimeOriginal                  Ascii      20  2004:06:21 23:37:53
http://0.0.0.0:1276/data/Reagan.tiff  Exif.Photo.DateTimeDigitized                 Ascii      20  2004:06:21 23:37:53
http://0.0.0.0:1276/data/Reagan.tiff  Iptc.Application2.City                       String     19  Straits of Magellan
http://0.0.0.0:1276/data/Reagan.tiff  Xmp.photoshop.City                           XmpText    19  Straits of Magellan
http://0.0.0.0:1276/data/exiv2-bug922a.jpg length = 7169623
s0                    Exif.Image.DateTime                          Ascii      20  2015:02:13 20:46:51
s0                    Exif.Photo.DateTimeOriginal                  Ascii      20  2015:02:13 13:51:35
s0                    Exif.Photo.DateTimeDigitized                 Ascii      20  2015:02:13 13:51:35
s1                    Exif.Image.DateTime                          Ascii      20  2015:02:13 20:46:51
s1                    Exif.Photo.DateTimeOriginal                  Ascii      20  2015:02:13 13:51:35
s1                    Exif.Photo.DateTimeDigitized                 Ascii      20  2015:02:13 13:51:35
s2                    Exif.Image.DateTime                          Ascii      20  2015:02:13 20:46:51
s2                    Exif.Photo.DateTimeOriginal                  Ascii      20  2015:02:13 13:51:35
s2                    Exif.Photo.DateTimeDigitized                 Ascii      20  2015:02:13 13:51:35
http://0.0.0.0:1276/data/exiv2-bug922a.jpg  Exif.Image.DateTime                          Ascii      20  2015:02:13 20:46:51
http://0.0.0.0:1276/data/exiv2-bug922a.jpg  Exif.Photo.DateTimeOriginal                  Ascii      20  2015:02:13 13:51:35
http://0.0.0.0:1276/data/exiv2-bug922a.jpg  Exif.Photo.DateTimeDigitized                 Ascii      20  2015:02:13 13:51:35
